core/src/main/java/org/apache/calcite/rel/metadata/RelMdUtil.java

```diff
@@ -659,15 +736,12 @@ public static Double getJoinRowCount(RelMetadataQuery mq, Join join,
     }
     double product = left * right;
 
-    // TODO:  correlation factor
     return product * mq.getSelectivity(join, condition);
   }
 
   /** Returns an estimate of the number of rows returned by a
    * {@link SemiJoin}. */
-  public static Double getSemiJoinRowCount(RelMetadataQuery mq, RelNode left,
-      RelNode right, JoinRelType joinType, RexNode condition) {
-    // TODO:  correlation factor
+  public static Double getSemiJoinRowCount(RelMetadataQuery mq, RelNode left, RexNode condition) {
```

Comment:
Why do you change the function signature? The right and joinType is not used for now, but doesn't mean we don't use it in the future. The current implementation is pretty naive.

Reply:
I agree, will add the unused arguments back.